worker/caasunitprovisioner/application_worker.go

```diff
@@ -169,15 +171,19 @@ func (aw *applicationWorker) loop() error {
 				appDeploymentWatcher = nil
 				continue
 			}
-			scale, err := aw.getScale()
-			if err != nil {
+			service, err := aw.serviceBroker.GetService(aw.application)
+			if err != nil && !errors.IsNotFound(err) {
 				return errors.Trace(err)
 			}
-			if scale == nil || *scale == lastReportedScale {
+			var scale *int
+			if service != nil {
+				scale = service.Scale
+			}
+			if scale != nil && *scale == lastReportedScale {
 				continue
 			}
 			lastReportedScale = *scale
```

Comment:
is it possible `scale == nil` here?

Reply:
Ah yes it is. We (now) need to continue to send status even if scale is nil but as you point out, I need to change the lastReportedScale logic.